### ISSUE TITLE
Knowledge Graph Sources?

### ISSUE THREAD

**Issue by logan-markewich** [jerryjliu/gpt_index#463]
Love the new support for knowledge graph triplets! I'm working on possibly augmenting the triplets with previously extracted triplets from dedicated methods (i.e. relation extraction, openIE).

Is there any way to show which triplets were used to generate the answer? Similar to how vector search keeps track of the source text?

I'm happy to make a PR for this if you think it's possible, just trying to sort out the existing code for myself right now.

**Comment by jerryjliu** [comment]
nice idea - we don't have support for this currently, but it seems pretty reasonable to augment the Response object with knowledge triplets.

if you had time, would love your help on a PR. I need to think a bit more on how this would work, but roughly speaking it could make sense to store the knowledge triplets in the `extra_info` in the Response object (Response defined in `gpt_index/response/schema.py`). 

If you look at the query though (`GPTKGTableQuery`), it only implements `_get_nodes_for_response`, which returns a list of nodes but doesn't provide an immediate avenue for also passing through the knowledge triplets. That's the main aspect to scope out

**Comment by logan-markewich** [comment]
Thanks for the tips! This will help a lot. I'll see if I can figure it out 👌

**Comment by logan-markewich** [comment]
@jerryjliu what do you think about implementing the `_query()` function in `GPTKGTableQuery` as well? Then we can easily access the `extra_info` from the text to add to the sources. The only downside is that the override is mostly the same code 🤔 The new code is in the else statement for each node in the returned tuples

Maybe something like this for `_qeury()`:
```
def _query(self, query_str: str) -> Response:
        """Answer a knowledge-graph query."""
        tuples = self.get_nodes_and_similarities_for_response(query_str)
        node_texts = []
        for node, similarity in tuples:
            text, response = self._get_text_from_node(query_str, node)
            self.response_builder.add_node(node, similarity=similarity)
            if response is not None:
                # these are source nodes from within this node (when it's an index)
                for source_node in response.source_nodes:
                    self.response_builder.add_source_node(source_node)
            else:
                    if 'rel_texts' in node.extra_info:
                        for rel_text in node.extra_info['rel_texts']:
                            self.response_builder.add_source_node(Node(text=rel_text))
            node_texts.append(text)

        <Same as base _query>
```

**Comment by jerryjliu** [comment]
yeah i think that makes sense! we don't have to use _get_nodes_for_response

**Comment by jerryjliu** [comment]
Should be closed with #472 !
